tests/integration/targets/nd_site/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+# Test code for the ND modules
+# Copyright: (c) 2023, Anvitha Jain (@anvitha-jain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# - name: Test that we have an Nexus Dashboard Insights Group and Site Name defined
+#   fail:
+#     msg: "Please define the following variables: insights_group, site_name."
+#   when: insights_group is not defined or site_name is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      output_level: '{{ site_output_level | default("debug") }}'
+
+
+# CLEAN ENVIRONMENT
```

Comment:
Not sure what is part of clean environment because you seem to also be testing querying and deleting. I would separate the logic of cleaning environment with your actual test cases.